src/System.Runtime.Analyzers/Core/SpecifyIFormatProvider.cs

```diff
@@ -27,42 +34,203 @@ public abstract class SpecifyIFormatProviderAnalyzer : DiagnosticAnalyzer
+                            {
+                                // Sample message 
+                                // 1. UICultureStringRule - 'TestClass.TestMethod()' passes 'Thread.CurrentUICulture' as the 'IFormatProvider' parameter to 'TestClass.CalleeMethod(string, IFormatProvider)'.
+                                // This property returns a culture that is inappropriate for formatting methods. If the result of 'TestClass.CalleeMethod(string, IFormatProvider)' will be displayed to the user,
+                                // specify 'CultureInfo.CurrentCulture' as the 'IFormatProvider' parameter. Otherwise, if the result will be stored and accessed by software, such as when it is persisted to disk
+                                // or to a database, specify 'CultureInfo.InvariantCulture'.
+                                // 2. UICultureRule -'TestClass.TestMethod()' passes 'CultureInfo.CurrentUICulture' as the 'IFormatProvider' parameter to 'TestClass.Callee(IFormatProvider, string)'.
+                                // This property returns a culture that is inappropriate for formatting methods. If the result of 'TestClass.Callee(IFormatProvider, string)' will be based on input from the user,
+                                // specify 'CultureInfo.CurrentCulture' as the 'IFormatProvider' parameter. Otherwise, if the result will based on input stored and accessed by software, such as when it is loaded
+                                // from disk or from a database, specify 'CultureInfo.InvariantCulture'.
+
+                                oaContext.ReportDiagnostic(
+                                    invocationExpression.Syntax.CreateDiagnostic(
+                                        targetMethod.ReturnType.Equals(stringType) ?
+                                            UICultureStringRule :
+                                            UICultureRule,
+                                        oaContext.ContainingSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                        symbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                        targetMethod.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                        currentCultureProperty.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                        invariantCultureProperty.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat)));
+                            }
+                        }
+                    }
+                    #endregion
+
+                }, OperationKind.InvocationExpression);
+            });
+        }
+
+        private IEnumerable<int> GetIndexesOfParameterType(IMethodSymbol targetMethod, INamedTypeSymbol formatProviderType)
+        {
+            return targetMethod.Parameters
+                .Select((Parameter, Index) => new { Parameter, Index })
```

Comment:
Don't capitalize local names. Curious how dows this work? Isn't targetMethod.Parameters just an array of parametersymbols? Where does index come from?